### PR TITLE
Mix and Mega: Expand mechanics

### PR DESCRIPTION
**Pull request by KrisXV**
https://www.smogon.com/forums/threads/putting-the-mix-back-into-mix-and-mega.3717633/

I'm opening this pull request early as a draft so others can look over the code just in case I missed something. Please don't merge this, I will merge it when the change is ready to go live.

### CODE REVIEW

File: data/mods/mixandmega/items.ts [hunk 1]
```diff
@@ -83,6 +114,15 @@ export const Items: {[k: string]: ModdedItemData} = {
...
 		isNonstandard: null,
 	},
+	griseouscore: {
+		inherit: true,
+		onBasePower(basePower, user, target, move) {
+			if (move.type === 'Ghost' || move.type === 'Dragon') {
+				return this.chainModify([4915, 4096]);
+			}
+		},
+		onTakeItem: false,
```
[the remark this answers is not in the document]
Comment (KrisXV) [reply data/mods/mixandmega/items.ts:124]: No, these are meant to be unreleased

**Comment by WeWuzNidokangz** [comment]
(Sorry, I left those on 'pending' when I looked before.)

File: data/mods/mixandmega/scripts.ts [hunk 2]
```diff
@@ -43,48 +427,53 @@ export const Scripts: ModdedBattleScriptsData = {
...
-			} else if (megaSpecies.types[1] !== baseSpecies.types[1]) {
-				deltas.type = megaSpecies.types[1];
+			} else if (formeChangeSpecies.types[1] !== baseSpecies.types[1]) {
+				deltas.type = formeChangeSpecies.types[1];
 			}
-			if (megaSpecies.isMega) deltas.isMega = true;
+			let formeType: string | null = null;
+			if (formeChangeSpecies.isMega) formeType = 'Mega';
+			if (formeChangeSpecies.isPrimal) formeType = 'Primal';
+			if (formeChangeSpecies.name.endsWith('Crowned')) formeType = 'Crowned';
```
[the remark this answers is not in the document]
Comment (KrisXV) [reply data/mods/mixandmega/scripts.ts:471]: This is pseudo handled already by not giving a formeType to the origin formes (and epilogue in capmnm) but if you want to make a PR for this feel free

**Comment by KrisXV** [comment]
> (Sorry, I left those on 'pending' when I looked before.)

no worries, better late than never!

File: server/chat-plugins/othermetas.ts [hunk 3]
```diff
@@ -247,19 +282,23 @@ export const commands: Chat.ChatCommands = {
 				Weight: (deltas.weighthg < 0 ? "" : "+") + deltas.weighthg / 10 + " kg",
 			};
 			let tier;
-			if (['redorb', 'blueorb'].includes(aStone.id)) {
+			if (['redorb', 'blueorb'].includes(aStone.id) || aStone.forcedForme?.endsWith('Origin')) {
```
[the remark this answers is not in the document]
Comment (KrisXV) [reply server/chat-plugins/othermetas.ts:287]: alternatively, calling them Crystals might work?

File: data/mods/mixandmega/scripts.ts [hunk 4]
```diff
@@ -8,6 +8,390 @@ export const Scripts: ModdedBattleScriptsData = {
...
+				const rawSpecies = this.actions.getMixedSpecies(pokemon.m.originalSpecies, item.forcedForme!, pokemon);
+				const species = pokemon.setSpecies(rawSpecies);
+				if (!species) continue;
+				pokemon.baseSpecies = rawSpecies;
+				pokemon.details = species.name + (pokemon.level === 100 ? '' : ', L' + pokemon.level) +
+					(pokemon.gender === '' ? '' : ', ' + pokemon.gender) + (pokemon.set.shiny ? ', shiny' : '');
+				pokemon.setAbility(species.abilities['0'], null, true);
+				pokemon.baseAbility = pokemon.ability;
+			}
+		}
```
Comment (Zarel) [inline data/mods/mixandmega/scripts.ts:74]: Okay, it looks like you're overriding `start` to put this here? Can this be moved to `onBegin`? It's very brittle to maintain a separate copy of the `start` script.

**Pull request merged**